groups/nts/ntsa/ntsa_endpoint.h

```diff
@@ -101,6 +102,10 @@ class Endpoint
     /// specified 'value'.
     explicit Endpoint(const ntsa::IpEndpoint& value);
 
+    /// Create a new endpoint having an "ip" representation having the
+    /// specified 'value'.
+    explicit Endpoint(const ntsa::Ipv4Endpoint& value);
```

Comment:
If we are going to do this, we might as well add explicit constructors for `ntsa::Ipv6Endpoint` and constructors for the "decomposition" of each of those types as well. I.e. also add an explicit constructor `explicit Endpoint(const ntsa::Ipv4Address& address, ntsa::Port port)` and `explicit Endpoint(const ntsa::Ipv6Address& address, ntsa::Port port)`.